Data is obtained from set of experimental crosses and data collected by Andrew Victory while an undergraduate at MSU in the Dworkin lab.

`readme.txt` for the genetic background effects of mutations influencing vein development.

 This serves as a summary of information associated with the file
`GeneticBackgroundVeinMutantInteractionsFinal.csv`

**BMB: you can include tables in an Rmarkdown file like this (or store such metadata in a separate file, if you're going to use it more than once)**
```{r metadata,echo=FALSE,message=FALSE, results="asis"}
md <- read.table(sep="|",header=TRUE, text=
"column      | meaning                                        
background.  | wild-type genetic background in which the crosses were made (2 backgrounds Ore & Sam)
gene1         | Mutations at first locus (in addition to rhomboid/rho)
gene2         | Mutations at Second locus (in addition to rhomboid/rho).
rhomboid      | Whether these individuals are homozygous for the rhomboid mutation (causing a shortening of the longitudinal veins).
rep.          | replicate blocks used in the experiment
sex           | sex of the fly being measured.
individual    | fly ID
L2-L5         | lengths of longitudinal veins
WL            | wing length"
)
knitr::kable(md)
```
### codes for backgrounds

- Ore: Oregon-R wild-type (marked with a w{-} eye colour marker)
- Sam: Samarkand wild-type (marked with a w{-} eye colour marker)

### codes for genes

- wt: wild-type
- rho: rhomboid (rho[KG07115]) 
- Egfr: Epidermal Growth Factor Receptor (Egfr[k05115])
- Bs: Blistered (bs[k07909] also called DSRF - Drosophila serum response factor)
- S: Star (S[k09530])
- Spi: spitz (spi[s3547])
- Aos:  argos (aos[W11])

### Other notes from the experiment:

- Ore; Bs/S; rho is empty because it was lethal in that combination. 
- The baseline strains (SAM w ; SAM; SAM rho & Ore w; ORE; ORE rho ) with rep1 is the "control" for those with an additional single 2nd Chr mutant. The baseline strains labeled rep1b (or rep2b) are for double mutants on the 2nd Chromosome. 
Sam baselines reps 3 and 4 are for comparison to any Oregon reps 3 or 4, as it was crossed in March.
- Rep1/Rep2: Original double and triple mutants, which now only includes SAM backgrounds (Crossed in October).
- Rep1b/Rep2b: Original Sam w; Sam; rho- from the set of triple mutants (done about 3 weeks after double mutants).
- Rep3/4: All Oregon double and triple mutants, and one set (in each rep) of Sam w; Sam; Sam for comparison to previous crosses (crossed in January).
- Rep3b/Rep4/b: Ore w; Ore; rho- from the set of triple mutants (although done at the same time as rep 3/4 single mutants, I did this for equal sets with Sam)

### Scientific questions: 

1. Distinguish what double or triple mutants have significant non-additive or epistatic effects. Meaning that double mutants have significantly shorter or longer vein length or wing length compared to a single rho mutant. 
2. Compare relationships as a function of genetic background. For example, does the type of genetic interaction or magnitude of genetic interaction change between genotypes in each background? 

### Analysis Plans:

1. Multivariate analysis: Observe genetic interactions in each vein.
2. Consider overall wing length as an effect on vein length. 

**BMB**: as always, we (JD and I) are going to give you a hard time whenever you state your scientific questions as "is there an effect?" or "does X affect Y?" or "is there a significant effect?" Statistical clarity (i.e. significance) is an important consideration -- if you don't achieve statistical clarity then you can't make any firm conclusions about what is happening in the biological system -- but it should never (almost never?) be part of your primary **scientific** question. Among other things, filtering by statistical significance gives rise to the "winner's curse" or "filter fallacy" <https://statmodeling.stat.columbia.edu/2010/10/02/the_winners_cur/>; "If an estimate is statistically significant, itâ€™s probably an overestimate of the magnitude of your effect". If you are screening mutants for follow-up studies, it would certainly be useful to pick out those that you can see clearly, but it is equally or more important to pick those of large effect (all other things being equal there **may** be a correlation between the magnitude of an effect and its statistical clarity, but you should never rely on this correlation to decide what's interesting). You might want to pick the mutants with the **largest** non-additive/epistatic effects; you might want to pick those that are most clearly large (i.e., those whose confidence limits fall the farthest from zero in *biological* (not Z-score) units). Non-additivity and epistasis are hard to put into clear biological terms, but you should try. 
